### PR TITLE
Move mark config to `src/mark.ts` + remove unnecessary namespace + mark specific config

### PR DESCRIPTION
**Pull request by kanitw**
Part of #1675 and #1727 

(This should be easily rebased into vg3)

Add mark specific config
- `config.mark.barBinSpacing/barThinSize/barSize` => `config.bar.binSpacing/continuousBandSize/discreteBandSize`
- `config.mark.lineSize` => `config.line.strokeWidth`
- `config.mark.shape` => `config.point.shape`
- `config.mark.size` => `config.point/circle/square.size`
- `config.mark.ruleSize` => `config.rule.strokeWidth`
- `config.mark.tickSize,tickThickness` => `config.tick.bandSize/thickness`
- `config.mark.*` (text properties) => `config.text.*`

### CODE REVIEW

File: src/compile/config.ts [hunk 1]
```diff
@@ -82,24 +89,24 @@ export function orient(mark: Mark, encoding: Encoding, scale: Dict<Scale>, markC
             yScaleType === ScaleType.ORDINAL) ||
             encoding.y.bin
           ) {
-        return Orient.VERTICAL;
+        return 'vertical';
```
Comment (domoritz) [inline src/compile/config.ts]: Do you like this better?

File: src/mark.ts [hunk 2]
```diff
@@ -27,3 +30,285 @@ export const SQUARE = Mark.SQUARE;
...
+
+export interface BarConfig extends MarkConfig {
+  /**
+   * Offset between bar for binned field.  Ideal value for this is either 0 (Preferred by statisticians) or 1 (Vega-Lite Default, D3 example style).
+   * @minimum 0
+   */
+  binSpacing?: number;
+  /**
+   * Default size of the bars on continuous scales.
+   */
```
Comment (domoritz) [inline src/mark.ts]: min 0

File: src/mark.ts [hunk 3]
```diff
@@ -27,3 +30,285 @@ export const SQUARE = Mark.SQUARE;
...
+  continuousBandSize?: number;
+
+  /**
+   * The size of the bars.  If unspecified, the default size is  `bandSize-1`,
+   * which provides 1 pixel offset between bars.
+   */
+  discreteBandSize?: number;
```
Comment (domoritz) [inline src/mark.ts]: min 0

File: src/mark.ts [hunk 4]
```diff
@@ -27,3 +30,285 @@ export const SQUARE = Mark.SQUARE;
...
+  /**
+   * The symbol shape to use. One of circle (default), square, cross, diamond, triangle-up, or triangle-down, or a custom SVG path.
+   */
+  shape?: Shape | string;
+
+  /**
+   * The pixel area each the point/circle/square.
+   * For example: in the case of circles, the radius is determined in part by the square root of the size value.
+   */
+  size?: number;
```
Comment (domoritz) [inline src/mark.ts]: min 0

File: src/mark.ts [hunk 5]
```diff
@@ -27,3 +30,285 @@ export const SQUARE = Mark.SQUARE;
...
+
+export interface TextConfig extends MarkConfig {
+  /**
+   * The horizontal alignment of the text. One of left, right, center.
+   */
+  align?: HorizontalAlign;
+  /**
+   * The rotation angle of the text, in degrees.
+   */
+  angle?: number;
```
Comment (domoritz) [inline src/mark.ts]: min 0 
max 360

File: src/mark.ts [hunk 6]
```diff
@@ -27,3 +30,285 @@ export const SQUARE = Mark.SQUARE;
...
+   */
+  theta?: number;
+  /**
+   * The typeface to set the text in (e.g., Helvetica Neue).
+   */
+  font?: string;
+  /**
+   * The font size, in pixels.
+   */
+  fontSize?: number;
```
Comment (domoritz) [inline src/mark.ts]: min 0

File: vega-lite-schema.json [hunk 7]
```diff
@@ -37,6 +37,72 @@
             ],
             "type": "string"
         },
+        "AreaConfig": {
```
Comment (domoritz) [inline vega-lite-schema.json]: I don't think we should regenerate the schema in a pr unless it is necessary so that tests pass

Comment (kanitw) [inline vega-lite-schema.json]: I disagree.  Schema diff is a great way to check if your PR / commit reports all the diff correctly (and we can copy the changes to release log easily.) 

Instead, I argue that we should always commit them in a PR.  Merge conflicts is easy to fix enough that it doesn't hurt to be consistent everytime.

Comment (domoritz) [inline vega-lite-schema.json]: Okay. Let's make that the strategy then.

**Comment by kanitw** [comment]
Okie.  @domoritz  -- I checked the gallery and added @minimum to all mark config.

File: src/mark.ts [hunk 8]
```diff
@@ -27,3 +30,306 @@ export const SQUARE = Mark.SQUARE;
...
 export const PRIMITIVE_MARKS = [AREA, BAR, LINE, POINT, TEXT, TICK, RULE, CIRCLE, SQUARE];
+
+export type FontStyle = 'normal' | 'italic';
+export type FontWeight = 'normal' | 'bold';
+/**
+ * @type integer
+ * @minimum 100
+ * @maximum 900
+ */
+export type FontWeightNumber = number;
```
Comment (domoritz) [inline src/mark.ts]: Which plugin?

Comment (kanitw) [inline src/mark.ts]: Oh the typescript => JSON schema

Comment (domoritz) [inline src/mark.ts]: I still don't understand what the problem is.

Comment (domoritz) [inline src/mark.ts]: I see. Maybe related to https://github.com/YousefED/typescript-json-schema/issues/57

**Pull request merged**